### ISSUE TITLE
v0.11.x breaking with docker compose profiles

### ISSUE THREAD

**Issue by ClayBenson94** [docker/buildx#1902]
### Contributing guidelines

- [X] I've read the [contributing guidelines](https://github.com/docker/buildx/blob/master/.github/CONTRIBUTING.md) and wholeheartedly agree

### I've found a bug and checked that ...

- [ ] ... the documentation does not mention anything about my problem
- [ ] ... there are no open or closed issues that are related to my problem

### Description

When building a simple docker-compose file that has a `profile`, buildx returns an error `failed to find target` on v0.11.0, but this isn't the case on v0.10.4

### Expected behaviour

A successful build of the target service

### Actual behaviour

I receive an error with `failed to find target` - details in the logs below

### Buildx version

github.com/docker/buildx v0.11.0 687feca9e8dcd1534ac4c026bc4db5a49de0dd6e

### Docker info

```text
Client:
 Context:    default
 Debug Mode: false
 Plugins:
  buildx: Docker Buildx (Docker Inc.)
    Version:  v0.10.4
    Path:     /Users/clay/.docker/cli-plugins/docker-buildx
  compose: Docker Compose (Docker Inc.)
    Version:  v2.17.3
    Path:     /Users/clay/.docker/cli-plugins/docker-compose
  dev: Docker Dev Environments (Docker Inc.)
    Version:  v0.1.0
    Path:     /Users/clay/.docker/cli-plugins/docker-dev
  extension: Manages Docker extensions (Docker Inc.)
    Version:  v0.2.19
    Path:     /Users/clay/.docker/cli-plugins/docker-extension
  init: Creates Docker-related starter files for your project (Docker Inc.)
    Version:  v0.1.0-beta.4
    Path:     /Users/clay/.docker/cli-plugins/docker-init
  sbom: View the packaged-based Software Bill Of Materials (SBOM) for an image (Anchore Inc.)
    Version:  0.6.0
    Path:     /Users/clay/.docker/cli-plugins/docker-sbom
  scan: Docker Scan (Docker Inc.)
    Version:  v0.26.0
    Path:     /Users/clay/.docker/cli-plugins/docker-scan
  scout: Command line tool for Docker Scout (Docker Inc.)
    Version:  v0.10.0
    Path:     /Users/clay/.docker/cli-plugins/docker-scout

Server:
 Containers: 0
  Running: 0
  Paused: 0
  Stopped: 0
 Images: 23
 Server Version: 23.0.5
 Storage Driver: overlay2
  Backing Filesystem: extfs
  Supports d_type: true
  Using metacopy: false
  Native Overlay Diff: true
  userxattr: false
 Logging Driver: json-file
 Cgroup Driver: cgroupfs
 Cgroup Version: 2
 Plugins:
  Volume: local
  Network: bridge host ipvlan macvlan null overlay
  Log: awslogs fluentd gcplogs gelf journald json-file local logentries splunk syslog
 Swarm: inactive
 Runtimes: io.containerd.runc.v2 runc
 Default Runtime: runc
 Init Binary: docker-init
 containerd version: 2806fc1057397dbaeefbea0e4e17bddfbd388f38
 runc version: v1.1.5-0-gf19387a
 init version: de40ad0
 Security Options:
  seccomp
   Profile: builtin
  cgroupns
 Kernel Version: 5.15.49-linuxkit
 Operating System: Docker Desktop
 OSType: linux
 Architecture: aarch64
 CPUs: 5
 Total Memory: 7.765GiB
 Name: docker-desktop
 ID: 67d7e666-a482-4020-b361-89c575c85a6c
 Docker Root Dir: /var/lib/docker
 Debug Mode: false
 HTTP Proxy: http.docker.internal:3128
 HTTPS Proxy: http.docker.internal:3128
 No Proxy: hubproxy.docker.internal
 Registry: https://index.docker.io/v1/
 Experimental: false
 Insecure Registries:
  hubproxy.docker.internal:5555
  127.0.0.0/8
 Live Restore Enabled: false
```

### Builders list

```text
NAME/NODE       DRIVER/ENDPOINT STATUS  BUILDKIT        PLATFORMS
default *       docker                                  
  default       default         running v0.10.6+d52b2d5 linux/arm64, linux/amd64, linux/amd64/v2, linux/riscv64, linux/ppc64le, linux/s390x, linux/386, linux/mips64le, linux/mips64, linux/arm/v7, linux/arm/v6
desktop-linux   docker                                  
  desktop-linux desktop-linux   running v0.10.6+d52b2d5 linux/arm64, linux/amd64, linux/amd64/v2, linux/riscv64, linux/ppc64le, linux/s390x, linux/386, linux/mips64le, linux/mips64, linux/arm/v7, linux/arm/v6
```

### Configuration

`Dockerfile`
```dockerfile
FROM scratch
```

`docker-compose-breaks.yml`
```yaml
---
version: '3.7'
services:
  some_service:
    profiles: ["test"]
    build:
      context: .
      dockerfile: Dockerfile
```

`docker-compose-works.yml`
```yaml
---
version: '3.7'
services:
  some_service:
    # profiles: ["test"]
    build:
      context: .
      dockerfile: Dockerfile
```

Steps to reproduce are identical for v0.10.4 and v0.11.0, but the former works, and latter errors

### Build logs

v0.11.0 (with profile)
```bash
./buildx-v0.11.0.darwin-arm64 bake --file docker-compose-breaks.yml some_service
[+] Building 0.0s (0/0)                                 docker:default
ERROR: failed to find target some_service
```

v0.10.4 (with profile)
```bash
$ docker buildx bake --file docker-compose-breaks.yml some_service
[+] Building 0.0s (3/3) FINISHED                                       
 => [internal] load .dockerignore                                 0.0s
 => => transferring context: 2B                                   0.0s
 => [internal] load build definition from Dockerfile              0.0s
 => => transferring dockerfile: 49B                               0.0s
 => exporting to image                                            0.0s
 => => writing image sha256:af863f3548d30332dd3a1952c4f764e0a09d  0.0s
```

v0.11.0 (no profile)
```bash
$ ./buildx-v0.11.0.darwin-arm64 bake --file docker-compose-works.yml some_service
[+] Building 0.0s (3/3) FINISHED                                                                                                                         docker:default
 => [internal] load .dockerignore                                                                                                                                  0.0s
 => => transferring context: 2B                                                                                                                                    0.0s
 => [internal] load build definition from Dockerfile                                                                                                               0.0s
 => => transferring dockerfile: 49B                                                                                                                                0.0s
 => exporting to image                                                                                                                                             0.0s
 => => writing image sha256:af863f3548d30332dd3a1952c4f764e0a09d5089be1746f3bfaff756633d5619                                                                       0.0s
```

v0.10.4 (no profile)
```bash
$ docker buildx bake --file docker-compose-works.yml some_service
[+] Building 0.0s (3/3) FINISHED                                                                                                                                        
 => [internal] load .dockerignore                                                                                                                                  0.0s
 => => transferring context: 2B                                                                                                                                    0.0s
 => [internal] load build definition from Dockerfile                                                                                                               0.0s
 => => transferring dockerfile: 49B                                                                                                                                0.0s
 => exporting to image                                                                                                                                             0.0s
 => => writing image sha256:af863f3548d30332dd3a1952c4f764e0a09d5089be1746f3bfaff756633d5619                                                                       0.0s
```

**Comment by basicdays** [comment]
We just ran into this as well. On Github Actions using [docker/setup-buildx-action](https://github.com/docker/setup-buildx-action) I think pulls in the latest version of buildx by default. Setting the action buildx version to `v0.10.5` has so far worked around this issue.

**Comment by Danysan1** [comment]
I ran into this as well from a Gitlab CI pipeline (described in [this stack overflow question](https://stackoverflow.com/questions/76505075/docker-buildx-bake-failed-to-find-target-error-only-in-ci)). While the issue is fixed, today I solved my problem by [imposing an older version of docker for the CI job](https://gitlab.com/openetymologymap/osm-wikidata-map-framework/-/commit/41f27f297c30fb74fad469b3e1d8136dd03e4902)

**Comment by crazy-max** [comment]
I think setting the env var `COMPOSE_PROFILES=*` should fix your issue while waiting for #1903 to be merged:

```
$ COMPOSE_PROFILES="*" docker buildx bake --file docker-compose-breaks.yml some_service
```

In GHA with the bake-action:

```yaml
      -
        name: Build and push
        uses: docker/bake-action@v3
        with:
          push: true
        env:
          COMPOSE_PROFILES: "*"
```
